test/solc/CommandLineInterface.cpp

```diff
@@ -1412,6 +1412,130 @@ BOOST_AUTO_TEST_CASE(cli_include_paths_ambiguous_import)
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ir-optimized", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") != std::string::npos);
+	// --ethdebug only works, if --via-ir was defined
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ethdebug", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ethdebug", "--via-ir", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("Debug Data (ethdebug):") != std::string::npos);
+	// --debug-info ethdebug always works with --strict-assembly
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--strict-assembly", tempDir.path().string() + "/input.yul"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") != std::string::npos);
+	// --debug-info ethdebug is excluded from --debug-info all
+	result = runCLI({"solc", "--debug-info", "all", "--strict-assembly", tempDir.path().string() + "/input.yul"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") == std::string::npos);
+}
+
+BOOST_AUTO_TEST_CASE(cli_ethdebug_ethdebug_output)
+{
+	TemporaryDirectory tempDir(TEST_CASE_NAME);
+	createFilesWithParentDirs({tempDir.path() / "input.sol"}, "pragma solidity >=0.0; contract C { function f() public pure {} }");
+	// --ethdebug only works with --via-ir
+	OptionsReaderAndMessages result = runCLI({"solc", "--ethdebug", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
+	result = runCLI({"solc", "--ethdebug", "--via-ir", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("Debug Data (ethdebug):") != std::string::npos);
+	result = runCLI({"solc", "--ethdebug", "--via-ir", "--ir", tempDir.path().string() + "/input.sol"}, "");
+	// --ethdebug will implicitly set --debug-info ethdebug, if no --debug-info was set.
+	// right now, if --debug-info ethdebug was set, we will output a comment on top of the ir output, indicating that ethdebug was enabled,
+	// so we use this to check whether --debug-info ethdebug was implicitly enabled.
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") != std::string::npos);
+	// let's just do the same for --ir-optimized
+	result = runCLI({"solc", "--ethdebug", "--via-ir", "--ir-optimized", "--optimize", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(result.success);
+	BOOST_REQUIRE(result.stdoutContent.find("/// ethdebug: enabled") != std::string::npos);
+	// --ethdebug only works with --via-ir.
+	result = runCLI({"solc", "--debug-info", "ethdebug", "--ethdebug", tempDir.path().string() + "/input.sol"}, "");
+	BOOST_REQUIRE(!result.success);
```

Comment:
These negative cases are already tested covered in other tests - they don't need to be here.